src/main/java/com/softdevsix/argos/service/RulesValidator.java

```diff
@@ -0,0 +1,78 @@
+package com.softdevsix.argos.service;
+
+import com.softdevsix.argos.domain.CodeComplexity;
+import com.softdevsix.argos.domain.CodeQuality;
+import com.softdevsix.argos.domain.CodeSmells;
+import com.softdevsix.argos.domain.Coverage;
+import com.softdevsix.argos.domain.Rules;
+import com.softdevsix.argos.domain.RulesRequestMap;
+
+public class RulesValidator {
+
+  public Rules validate(RulesRequestMap rulesRequestMap){
+    Rules rules = new Rules();
+    validateAndSetCodeQuality(rulesRequestMap, rules );
+    validateAndSetBestPractices(rulesRequestMap, rules );
+    validateAndSetCodeSmells(rulesRequestMap, rules );
+    validateAndSetCodeComplexity(rulesRequestMap, rules );
+    validateAndSetCodingStandards(rulesRequestMap, rules );
+    validateAndSetCoverage(rulesRequestMap, rules );
+    return rules;
+  }
+
+  private void validateAndSetCodeQuality(RulesRequestMap rulesRequestMap, Rules validatedRules) {
+    if (rulesRequestMap.getCodeQuality() != null) {
+      CodeQuality codeQuality = rulesRequestMap.getCodeQuality();
+      if (codeQuality.isMaxLineLengthEnabled() && codeQuality.getMaxLineLengthLimit() <= 0) {
+        throw new IllegalArgumentException("The line length limit must be positive");
+      }
+      validatedRules.setCodeQuality(codeQuality);
+    }
+  }
+
+  private void validateAndSetBestPractices(RulesRequestMap rulesRequestMap, Rules validatedRules) {
+    if (rulesRequestMap.getBestPractices() != null) {
+      validatedRules.setBestPractices(rulesRequestMap.getBestPractices());
```

Comment:
please never ever do this again, since this is a void and we are setting a value inside this void, it will become a difficult part to test, it is better to return the object in this case the Rules object